// @agm/core/directives/map.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@agm/core/services/managers/marker-manager");
var i2 = require("@agm/core/services/google-maps-api-wrapper");
var i3 = require("@agm/core/services/managers/info-window-manager");
var i4 = require("@agm/core/services/managers/circle-manager");
var i5 = require("@agm/core/services/managers/polyline-manager");
var i6 = require("@agm/core/services/managers/polygon-manager");
var i7 = require("@agm/core/services/managers/kml-layer-manager");
var i8 = require("@agm/core/services/managers/data-layer-manager");
var i9 = require("@agm/core/services/maps-api-loader/maps-api-loader");
var i10 = require("@agm/core/directives/map");
var styles_AgmMap = ['.agm-map-container-inner[_ngcontent-%COMP%] {\n      width: inherit;\n      height: inherit;\n    }\n    .agm-map-content[_ngcontent-%COMP%] {\n      display:none;\n    }'];
exports.RenderType_AgmMap = i0.ɵcrt({ encapsulation: 0, styles: styles_AgmMap,
    data: {} });
function View_AgmMap_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 0, 'div', [['class', 'agm-map-container-inner sebm-google-map-container-inner']], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'agm-map-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n      '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵted(null, ['\n  ']))], null, null);
}
exports.View_AgmMap_0 = View_AgmMap_0;
function View_AgmMap_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 9, 'agm-map', [], [[2, 'sebm-google-map-container', null]], null, null, View_AgmMap_0, exports.RenderType_AgmMap)), i0.ɵprd(4608, null, i1.MarkerManager, i1.MarkerManager, [i2.GoogleMapsAPIWrapper, i0.NgZone]), i0.ɵprd(4608, null, i3.InfoWindowManager, i3.InfoWindowManager, [i2.GoogleMapsAPIWrapper, i0.NgZone,
            i1.MarkerManager]), i0.ɵprd(4608, null, i4.CircleManager, i4.CircleManager, [i2.GoogleMapsAPIWrapper, i0.NgZone]), i0.ɵprd(4608, null, i5.PolylineManager, i5.PolylineManager, [i2.GoogleMapsAPIWrapper, i0.NgZone]), i0.ɵprd(4608, null, i6.PolygonManager, i6.PolygonManager, [i2.GoogleMapsAPIWrapper, i0.NgZone]), i0.ɵprd(4608, null, i7.KmlLayerManager, i7.KmlLayerManager, [i2.GoogleMapsAPIWrapper,
            i0.NgZone]), i0.ɵprd(4608, null, i8.DataLayerManager, i8.DataLayerManager, [i2.GoogleMapsAPIWrapper, i0.NgZone]), i0.ɵprd(512, null, i2.GoogleMapsAPIWrapper, i2.GoogleMapsAPIWrapper, [i9.MapsAPILoader, i0.NgZone]), i0.ɵdid(770048, null, 0, i10.AgmMap, [i0.ElementRef, i2.GoogleMapsAPIWrapper], null, null)], function (_ck, _v) {
        _ck(_v, 9, 0);
    }, function (_ck, _v) {
        var currVal_0 = true;
        _ck(_v, 0, 0, currVal_0);
    });
}
exports.View_AgmMap_Host_0 = View_AgmMap_Host_0;
exports.AgmMapNgFactory = i0.ɵccf('agm-map', i10.AgmMap, View_AgmMap_Host_0, { longitude: 'longitude', latitude: 'latitude', zoom: 'zoom', minZoom: 'minZoom',
    maxZoom: 'maxZoom', draggable: 'mapDraggable', disableDoubleClickZoom: 'disableDoubleClickZoom',
    disableDefaultUI: 'disableDefaultUI', scrollwheel: 'scrollwheel', backgroundColor: 'backgroundColor',
    draggableCursor: 'draggableCursor', draggingCursor: 'draggingCursor', keyboardShortcuts: 'keyboardShortcuts',
    zoomControl: 'zoomControl', zoomControlOptions: 'zoomControlOptions', styles: 'styles',
    usePanning: 'usePanning', streetViewControl: 'streetViewControl', streetViewControlOptions: 'streetViewControlOptions',
    fitBounds: 'fitBounds', scaleControl: 'scaleControl', scaleControlOptions: 'scaleControlOptions',
    mapTypeControl: 'mapTypeControl', mapTypeControlOptions: 'mapTypeControlOptions',
    panControl: 'panControl', panControlOptions: 'panControlOptions', rotateControl: 'rotateControl',
    rotateControlOptions: 'rotateControlOptions', fullscreenControl: 'fullscreenControl',
    fullscreenControlOptions: 'fullscreenControlOptions', mapTypeId: 'mapTypeId',
    clickableIcons: 'clickableIcons', gestureHandling: 'gestureHandling' }, { mapClick: 'mapClick',
    mapRightClick: 'mapRightClick', mapDblClick: 'mapDblClick', centerChange: 'centerChange',
    boundsChange: 'boundsChange', idle: 'idle', zoomChange: 'zoomChange', mapReady: 'mapReady' }, ['*']);
